libcontainer/intelrdt/intelrdt.go

```diff
@@ -589,6 +600,28 @@ func (m *Manager) GetStats() (*Stats, error) {
 	return stats, nil
 }
 
+func combineSchemas(l3CacheSchema, memBwSchema string) string {
+	// If both l3CacheSchema and memBwSchema are set and
+	// l3CacheSchema contains a line beginning with "MB:", the
+	// value written to schemata file MUST be the non-"MB:"
+	// line(s) from l3CacheSchema and the line from memBWSchema.
+
+	validLines := make([]string, 0)
+
+	// Split the l3CacheSchema to lines.
+	lines := strings.Split(l3CacheSchema, "\n")
+
+	// Remove the "MB:" lines.
+	for _, line := range lines {
+		if strings.HasPrefix(line, "MB:") {
+			continue
+		}
+		validLines = append(validLines, line)
+	}
+
+	return strings.Join(validLines, "\n") + "\n" + memBwSchema
```

Comment:
This is not very efficient in terms of resources. We split the strings and then we combine them, creating two slices in the process. I'm not sure if there's a better solution (except than a slight optimization of reusing the same slice -- see https://github.com/golang/go/wiki/SliceTricks#filtering-without-allocating).